pennylane/ops/op_math/sprod.py

```diff
@@ -250,3 +250,13 @@ def simplify(self) -> Operator:
         if isinstance(new_base, SProd):
             return SProd(scalar=self.scalar, base=new_base).simplify()
         return SProd(scalar=self.scalar, base=new_base)
+
+    @property
+    def hash(self):
+        return hash(
+            (
+                str(self.name),
+                self.base.hash,
+                self.scalar,
```

Comment:
Careful:
```pycon
>>> qml.s_prod(np.array(2), qml.PauliX(0)).hash
TypeError: unhashable type: 'numpy.ndarray'
```

Reply:
One solution would be to cast it to a string.

Reply:
Done